hail/src/main/scala/is/hail/io/fs/HadoopFS.scala

```diff
@@ -120,19 +121,19 @@ class HadoopFS(private[this] var conf: SerializableHadoopConfiguration) extends
     }
   }
 
-  override def mkDir(dirname: String): Unit = {
+  override def mkDir(dirname: String): Unit = retryTransientErrors {
     getFileSystem(dirname).mkdirs(new hadoop.fs.Path(dirname))
   }
 
-  def remove(fname: String): Unit = {
+  def remove(fname: String): Unit = retryTransientErrors {
     getFileSystem(fname).delete(new hadoop.fs.Path(fname), false)
```

Comment:
For retry transient to work you'll need to ensure idempotency. I think it's sufficient to try-catch and smother FileNotFoundException Same for delete. Not sure about rmtree. You'll have to try calling that on a directory that doesn't exist.